src/resources/parser/glb-parser.js

```diff
@@ -298,24 +388,120 @@ Object.assign(pc, function () {
+            var primitiveType, vertexBuffer, numIndices;
+            var indices = null;
+            var mesh = new pc.Mesh();
 
-            var vertexBuffer = createVertexBuffer(device, attributes, indices, accessors, bufferViews, buffers);
+            // try and get draco compressed data first
+            if (primitive.hasOwnProperty('extensions')) {
+                var extensions = primitive.extensions;
+                if (extensions.hasOwnProperty('KHR_draco_mesh_compression')) {
+
+                    // access DracoDecoderModule
+                    var decoderModule = window.DracoDecoderModule;
+                    if (decoderModule) {
+                        var extDraco = extensions.KHR_draco_mesh_compression;
+                        if (extDraco.hasOwnProperty('attributes')) {
+                            var bufferView = bufferViews[extDraco.bufferView];
+                            var arrayBuffer = buffers[bufferView.buffer];
+                            var uint8Buffer = new Uint8Array(arrayBuffer.buffer, arrayBuffer.byteOffset + bufferView.byteOffset, bufferView.byteLength);
+                            var buffer = new decoderModule.DecoderBuffer();
+                            buffer.Init(uint8Buffer, uint8Buffer.length);
+
+                            var decoder = new decoderModule.Decoder();
+                            var geometryType = decoder.GetEncodedGeometryType(buffer);
+
+                            var outputGeometry, status;
+                            switch (geometryType) {
+                                case decoderModule.POINT_CLOUD:
+                                    primitiveType = pc.PRIMITIVE_POINTS;
+                                    outputGeometry = new decoderModule.PointCloud();
+                                    status = decoder.DecodeBufferToPointCloud(buffer, outputGeometry);
+                                    break;
+                                case decoderModule.TRIANGULAR_MESH:
+                                    primitiveType = pc.PRIMITIVE_TRIANGLES;
+                                    outputGeometry = new decoderModule.Mesh();
+                                    status = decoder.DecodeBufferToMesh(buffer, outputGeometry);
+                                    break;
+                                default:
+                                case decoderModule.INVALID_GEOMETRY_TYPE:
+                                    console.error('Mesh asset - invalid draco compressed geometry type: ' + geometryType);
+                                    break;
+                            }
 
-            var mesh = new pc.Mesh();
+                            if (!status.ok() || outputGeometry.ptr == 0) {
+                                // #ifdef DEBUG
+                                console.error("Failed to decode draco compressed asset: " + status.error_msg());
+                                // #endif
```

Comment:
should the draco code below continue to be executed if this fails? shouldn't this result in load failure?

Reply:
done